lib/commands/config/generate.ts

```diff
@@ -192,6 +194,7 @@ export default class ConfigGenerateCmd extends Command {
 		});
 		answers.version = options.version;
 		answers.provisioningKeyName = options['provisioning-key-name'];
+		answers.developmentMode = options.dev;
```

Comment:
Setting `answers.developmentMode` here causes this option to be eventually passed to:
`sdk.models.os.getConfig(fleetSlug, { developmentMode: true/false, ... })`
(in `lib/utils/config.ts` that did not need to be modified by this PR).

I have tested it, it seems to work OK.

Reply:
We probably need to add a check to throw an error if the user is mixing things like: `--version x.y.z+revN.prod --dev`.
Using balenaSemver to get check the build metadata parts for `prod` should do the trick.

Reply:
For this specific command we do not need to resolve the version and it's find as is :+1: (since non complete versions are acceptable be the API and it's mostly for finding out what type of API keys the OS supports)